agent/engine/daemonmanager/daemon_manager_linux.go

```diff
@@ -17,191 +17,14 @@
 package daemonmanager
 
 import (
-	"context"
-	"encoding/json"
-	"fmt"
 	"io/fs"
-	"os"
-	"path/filepath"
-
-	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
-	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
-	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
-	"github.com/aws/amazon-ecs-agent/agent/taskresource"
-	utils "github.com/aws/amazon-ecs-agent/agent/utils"
-	"github.com/aws/amazon-ecs-agent/agent/utils/loader"
-	apicontainerstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/container/status"
-	apitaskstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/task/status"
-	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
-	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/field"
-
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/docker/docker/api/types"
-	dockercontainer "github.com/docker/docker/api/types/container"
-	dockermount "github.com/docker/docker/api/types/mount"
-	"github.com/pborman/uuid"
 )
 
 const (
-	daemonUID                                   = 0
-	daemonMountPermission           fs.FileMode = 0755
-	daemonLogPermission             fs.FileMode = 0777
-	ecsAgentLogFileENV                          = "ECS_LOGFILE"
-	defaultECSAgentLogPathContainer             = "/log"
-	socketPathHostRoot                          = "/var/run/ecs"
-	logPathHostRoot                             = "/log/daemons"
+	daemonUID                                     = 0
```

Comment:
nit (non-blocking): I'm seeing a lot of different config files that's scattered in different files. Perhaps we can try to scope it down a bit to having fewer config files.

Reply:
I have refactored as much as possible. For some constants that are specific to a module, I have kept it to the module itself. Most are limited to the module. For the Linux ones, I refactored as much as possible, but didn't want to perturb the code too much. Let me know if you see any config that can be made global.